pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -801,21 +853,12 @@ pub mod pallet {
 				Error::<T>::RouterNotFound
 			);
 
-			let nonce = <OutboundMessageNonceStore<T>>::try_mutate(|n| {
-				n.ensure_add_assign(T::OutboundMessageNonce::one())?;
-				Ok::<T::OutboundMessageNonce, DispatchError>(*n)
-			})?;
-
-			OutboundMessageQueue::<T>::insert(
-				nonce,
-				(destination.clone(), T::Sender::get(), message.clone()),
-			);
-
-			Self::deposit_event(Event::OutboundMessageSubmitted {
-				sender: T::Sender::get(),
-				domain: destination,
-				message,
-			});
+			match PackedMessage::<T>::get((&sender, &destination)) {
+				Some(packed) => {
+					PackedMessage::<T>::insert((sender, destination), packed.pack(message)?)
```

Comment:
@mustermeiszer Here the `pack()` checks that you don't exceed the batch limit, so you can never create more than 16 messages in a batch. Creating the 17th will fail. `on_idle` just will send the bytes.

Reply:
Cool! We still need to check 
* what is the weight of a 16 message batch?
* do we need to adapt `process_message` weight defaults